src/views/DataExploration/components/TreeFacet/TreeFacetModal.tsx

```diff
@@ -0,0 +1,361 @@
+import { ReactElement, useEffect, useRef, useState } from 'react';
```

Comment:
I think it is legacy code. If not, I would try to:

- breakdown the file (relatively large)
- be a bit more functional (that s a personal taste, but a paradigm we use quite a bit)

Reply:
It's legacy, I prefer to not touch it